lib/compat.php

```diff
@@ -261,46 +261,99 @@ function gutenberg_add_date_settings_timezone( $scripts ) {
+			'l10n'     => array(
+				'locale'        => get_user_locale(),
+				'months'        => array_values( $wp_locale->month ),
+				'monthsShort'   => array_values( $wp_locale->month_abbrev ),
+				'weekdays'      => array_values( $wp_locale->weekday ),
+				'weekdaysShort' => array_values( $wp_locale->weekday_abbrev ),
+				'meridiem'      => (object) $wp_locale->meridiem,
+				'relative'      => array(
+					/* translators: %s: Duration. */
+					'future' => __( '%s from now', 'default' ),
+					/* translators: %s: Duration. */
+					'past'   => __( '%s ago', 'default' ),
+				),
+			),
+			'formats'  => array(
+				/* translators: Time format, see https://www.php.net/date */
+				'time'                => get_option( 'time_format', __( 'g:i a', 'default' ) ),
+				/* translators: Date format, see https://www.php.net/date */
+				'date'                => get_option( 'date_format', __( 'F j, Y', 'default' ) ),
+				/* translators: Date/Time format, see https://www.php.net/date */
+				'datetime'            => __( 'F j, Y g:i a', 'default' ),
+				/* translators: Abbreviated date/time format, see https://www.php.net/date */
+				'datetimeAbbreviated' => __( 'M j, Y g:i a', 'default' ),
+			),
+			'timezone' => array(
+				'offset' => get_option( 'gmt_offset', 0 ),
+				'string' => $timezone_string,
+				'abbr'   => $timezone_abbr,
+			),
+		)
+	);
+
 	$scripts->add_inline_script(
 		'wp-date',
 		sprintf(
 			'wp.date.setSettings( %s );',
-			wp_json_encode(
-				array(
-					'l10n'     => array(
-						'locale'        => get_user_locale(),
-						'months'        => array_values( $wp_locale->month ),
-						'monthsShort'   => array_values( $wp_locale->month_abbrev ),
-						'weekdays'      => array_values( $wp_locale->weekday ),
-						'weekdaysShort' => array_values( $wp_locale->weekday_abbrev ),
-						'meridiem'      => (object) $wp_locale->meridiem,
-						'relative'      => array(
-							/* translators: %s: Duration. */
-							'future' => __( '%s from now', 'default' ),
-							/* translators: %s: Duration. */
-							'past'   => __( '%s ago', 'default' ),
-						),
-					),
-					'formats'  => array(
-						/* translators: Time format, see https://www.php.net/date */
-						'time'                => get_option( 'time_format', __( 'g:i a', 'default' ) ),
-						/* translators: Date format, see https://www.php.net/date */
-						'date'                => get_option( 'date_format', __( 'F j, Y', 'default' ) ),
-						/* translators: Date/Time format, see https://www.php.net/date */
-						'datetime'            => __( 'F j, Y g:i a', 'default' ),
-						/* translators: Abbreviated date/time format, see https://www.php.net/date */
-						'datetimeAbbreviated' => __( 'M j, Y g:i a', 'default' ),
-					),
-					'timezone' => array(
-						'offset' => get_option( 'gmt_offset', 0 ),
-						'string' => $timezone_string,
-						'abbr'   => $timezone_abbr,
-					),
-				)
-			)
+			$default_settings
 		),
 		'after'
 	);
+
+	$scripts->add_inline_script(
```

Comment:
Can we add a `deprecated` call each in this inline script to clarify that using "moment" is entirely deprecated?